contracts/src/v0.8/ccip/DonIDClaimer.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {Ownable2StepMsgSender} from "../shared/access/Ownable2StepMsgSender.sol";
+import {ITypeAndVersion} from "../shared/interfaces/ITypeAndVersion.sol";
+
+interface ICapabilitiesRegistry {
+  /// @notice Gets the next available DON ID from the CapabilitiesRegistry
+  /// @return uint32 The next available DON ID
+  function getNextDONId() external view returns (uint32);
+}
+
+/// @notice DonIDClaimer contract is used to claim and manage DON IDs. It interacts with
+/// the CapabilitiesRegistry to fetch the next available DON ID and allows
+/// for synchronization of the DON ID with an optional offset to skip certain
+/// DON IDs as needed. The contract provides functionality for claiming,
+/// retrieving, and syncing DON IDs, ensuring that multiple workflows can
+/// manage DON IDs without conflict or accidental reuse.
+/// @dev The contract maintains its own internal counter for DON IDs and ensures
+/// the next available ID is claimed and tracked by the contract. The sync function
+/// allows for alignment with the CapabilitiesRegistry.
+contract DonIDClaimer is ITypeAndVersion, Ownable2StepMsgSender {
+  error ZeroAddressNotAllowed();
+  error AccessForbidden(address sender);
+
+  event AuthorizedDeployerSet(address indexed senderAddress, bool allowed);
+  event DonIDClaimed(address indexed claimer, uint32 donId);
+  event DonIDSynced(uint32 newDONId);
+
+  string public constant override typeAndVersion = "DonIDClaimer 1.0.0-dev";
+  /// @notice The next available DON ID that is claimed and incremented
+  uint32 private s_nextDONId;
+
+  /// @notice The address of the CapabilitiesRegistry contract used to fetch the next DON ID
+  ICapabilitiesRegistry private immutable i_capabilitiesRegistry;
+
+  /// @notice Mapping to track authorized deployed keys
+  mapping(address => bool) private s_authorizedDeployers;
```

Comment:
Please use Openzeppelin enumerableSet and add a getter func that returns all deployers